Model/Api/GetProduct.php

```diff
@@ -65,60 +91,120 @@ class GetProduct implements GetProductInterface
 
     /**
      * @param ProductRepositoryInterface  $productRepositoryInterface
-     * @param StockRegistryInterface  $stockRegistry
+     * @param StockRegistryProviderInterface  $stockRegistry
+     * @param StockConfigurationInterface  $stockConfiguration
      * @param GetProductDataInterface  $productData
      * @param StoreManagerInterface $storeManager
+     * @param Configurable          $configurable
      * @param HookHelper            $hookHelper
      * @param Bugsnag               $bugsnag
      */
     public function __construct(
         ProductRepositoryInterface  $productRepositoryInterface,
-        StockRegistryInterface  $stockRegistry,
+        StockRegistryProviderInterface  $stockRegistry,
+        StockConfigurationInterface  $stockConfiguration,
         GetProductDataInterface  $productData,
         StoreManagerInterface $storeManager,
+        Configurable          $configurable,
         HookHelper $hookHelper,
         Bugsnag $bugsnag
     ) {
         $this->productRepositoryInterface = $productRepositoryInterface;
         $this->stockRegistry = $stockRegistry;
+        $this->stockConfiguration = $stockConfiguration;
         $this->productData = $productData;
         $this->storeManager = $storeManager;
         $this->hookHelper = $hookHelper;
         $this->bugsnag = $bugsnag;
+        $this->configurable = $configurable;
+    }
+
+    private function getStockStatus($product){
```

Comment:
@danac-gs had to add this function for backwards compadibilty. Would love your thoughts on this.

I followed this article https://www.mexbs.com/magento-blog/getting-the-stock-status-programatically-in-magento-2-with-and-without-msi/

Reply:
Merging now to test, will circle back to this comment in seperate PR